lib/rubocop/rspec/language.rb

```diff
@@ -2,152 +2,200 @@
+        def self.all(element)
+          Language.config['HookScopes'].include?(element.to_s)
+        end
       end
 
-      module Examples
-        EXAMPLES = SelectorSet.new(%i[it specify example scenario its])
-        FOCUSED  = SelectorSet.new(%i[fit fspecify fexample fscenario focus])
-        SKIPPED  = SelectorSet.new(%i[xit xspecify xexample xscenario skip])
-        PENDING  = SelectorSet.new(%i[pending])
+      module Includes # :nodoc: all
+        def self.all(element)
+          Example.all(element) ||
+            Context.all(element)
+        end
 
-        ALL = EXAMPLES + FOCUSED + SKIPPED + PENDING
-      end
+        module Example
+          def self.all(element)
+            Language.config['Includes']['Example'].include?(element.to_s)
+          end
+        end
 
-      module Hooks
-        ALL = SelectorSet.new(
-          %i[
-            prepend_before
-            before
-            append_before
-            around
-            prepend_after
-            after
-            append_after
-          ]
-        )
-
-        module Scopes
-          ALL = SelectorSet.new(
-            %i[
-              each
-              example
-              context
-              all
-              suite
-            ]
-          )
+        module Context
+          def self.all(element)
+            Language.config['Includes']['Context'].include?(element.to_s)
+          end
         end
       end
 
-      module Helpers
-        ALL = SelectorSet.new(%i[let let!])
+      module Runners # :nodoc:
+        def self.all(element)
+          Language.config['Runners'].include?(element.to_s)
+        end
       end
 
-      module Subject
-        ALL = SelectorSet.new(%i[subject subject!])
+      module SharedGroups # :nodoc: all
+        def self.all(element)
+          Example.all(element) ||
+            Context.all(element)
+        end
+
+        module Example
+          def self.all(element)
+            Language.config['SharedGroups']['Example'].include?(element.to_s)
+          end
+        end
+
+        module Context
+          def self.all(element)
+            Language.config['SharedGroups']['Context'].include?(element.to_s)
+          end
+        end
       end
 
-      module Expectations
-        ALL = SelectorSet.new(%i[expect is_expected expect_any_instance_of])
+      module Subjects # :nodoc:
+        def self.all(element)
+          Language.config['Subjects'].include?(element.to_s)
+        end
       end
 
-      module Runners
-        ALL = SelectorSet.new(%i[to to_not not_to])
+      # This is used in Dialect and DescribeClass cops to detect RSpec blocks
+      module ALL # :nodoc:
+        def self.all(element)
+          ExampleGroups.all(element) ||
+            Examples.all(element) ||
+            Helpers.all(element) ||
+            Hooks.all(element) ||
+            Includes.all(element) ||
+            SharedGroups.all(element) ||
+            Subjects.all(element)
```

Comment:
I've left out `Expectations.all` and `Runners.all` from here. They don't seem to be needed for our cops (`Dialect` and `DescribeClass`), and it triggered Cyclomatic Complexity cop.